src/App.tsx

```diff
@@ -1,300 +1,100 @@
+import ColorPicker from "./components/color-picker/ColorPicker";
+import SavePaletteSection from "./sections/save-palette-section/SavePaletteSection";
+import PromptDialog from "./components/prompt-dialog/PromptDialog";
+import AppController from "./app.controller";
 
-  const [isEditNamePalette, setIsEditNamePalette] = useState("");
-  const [valueEditNamePalette, setValueEditNamePalette] = useState("");
-  const handledEditNamePalette = (name: string) => {
-    setIsEditNamePalette(name);
-    setValueEditNamePalette(name);
-  };
 
-  const savedPalettes = useSyncExternalStore<Palettes>(
-    store.subscribe,
-    store.getSnapshot
-  );
+const controller: AppController = new AppController();
 
-  const [, setLocation] = useLocation();
+function App() {
+  const provider = useContext(HexadecimalContext);
 
-  const [showPromptDialog, setShowPromptDialog] = useState(false);
-  const [isGenerating, setIsGenerating] = useState(false);
+  const [ , setLocation ] = useLocation();
 
-  const colors = useMemo(() => {
-    return chroma
-      .scale(["#FFFFFF", deferredColor, "#000000"])
-      .colors(11)
-      .slice(1, 10)
-      .map((color) => {
-        return {
-          color,
-          text: chroma.contrast(color, "#191919") > 4.5 ? "#191919" : "#FEFDFC",
-        };
-      });
-  }, [deferredColor, colorAux]);
+  const [ color, setColor ] = useState("#ffffff");
 
-  const isValid = useCallback((newColor: string) => {
-    const regex = /^#?([a-fA-F0-9]{6}|[a-fA-F0-9]{3})$/;
+  const [ showPromptDialog, setShowPromptDialog ] = useState(false);
+  const [ isGenerating, setIsGenerating ] = useState(false);
 
-    if (regex.test(newColor)) {
-      setColor(newColor);
-      setColorAux(newColor);
-      setLocation("?color=%23" + newColor.slice(1, 8));
-      return true;
-    }
-    return false;
-  }, []);
+  const colors = useMemo(() => controller.getColor(color), [ color ]);
 
   useEffect(() => {
-    const search = window.location.search;
-    const queryString = new URLSearchParams(search).get("color");
-    if (!queryString) {
-      handleGenerateRandom();
-    } else {
-      const newColor = queryString;
-      isValid(newColor);
-    }
-  }, []);
-
-  const handleColorChange = useCallback(
-    (event: ChangeEvent<HTMLInputElement>) => {
-      const newColor = event.target.value;
-
-      debounce({
-        callback: () => isValid(newColor),
-      });
-    },
-    []
-  );
+    if (!ValidateHexadecimal(provider.hexColor)) return;
 
-  const handleTextInputChange = (event: ChangeEvent<HTMLInputElement>) => {
-    const newColor = event.target.value;
-    if (newColor.length > 7) return;
-    setColorAux(newColor);
-    isValid(newColor);
-  };
+    setColor(provider.hexColor);
 
-  const handleGenerateRandom = () => {
-    const newColor = getRandomColor();
-    isValid(newColor);
-  };
+    setLocation("?color=%23" + provider.hexColor.slice(1, 8));
+  }, [ provider.hexColor, setLocation ]);
```

Comment:
What if we got rid of this by updating the location within the HexadecimalContext provider, as part of the decimal change handler?

Also, do we really need to keep `color` in state if it's already part of the context we're reading?

Reply:
Super valid point! ❤️

Reply:
(Also, sorry about that, it's a bit hard to determine what was contributed by whom at this point given this is the first time I look at this code outside of stream)